app/models/provider.rb

```diff
@@ -119,6 +122,16 @@ def user_url
     ENV["VOLPINO_URL"] + "/users?provider-id=" + symbol.downcase
   end
 
+  def self.push_to_index
+    data = self.all
```

Comment:
I would avoid `self.all` together with `.each`. You would load everything into memory and this does not scale well. Better to use `find_each`: https://apidock.com/rails/ActiveRecord/Batches/find_each